gridsite-1.5-compat package.

diff --git a/org.gridsite.core/src/Makefile b/org.gridsite.core/src/Makefile
index 975d5d7..067cb8d 100644
--- a/org.gridsite.core/src/Makefile
+++ b/org.gridsite.core/src/Makefile
@@ -89,6 +89,10 @@ libgridsite.so.$(VERSION): grst_err.o grst_x509.o grst_gacl.o grst_xacml.o \
          grst_asn1.o grst_htcp.o 
 	ln -sf libgridsite.so.$(VERSION) libgridsite.so
 	ln -sf libgridsite.so.$(VERSION) libgridsite.so.$(MINOR_VERSION)
+	gcc -shared -Wl,-soname,libgridsite.so.1.5 \
+         -o libgridsite.so.1.5 -lcrypto `xml2-config --libs` \
+         grst_err.o grst_x509.o grst_gacl.o grst_xacml.o grst_http.o \
+         grst_asn1.o grst_htcp.o
 
 libgridsite.a: grst_err.o grst_x509.o grst_gacl.o grst_xacml.o grst_http.o grst_asn1.o grst_htcp.o
 	ar src libgridsite.a grst_err.o grst_x509.o grst_gacl.o grst_xacml.o grst_http.o grst_asn1.o grst_htcp.o
@@ -134,6 +138,10 @@ libgridsite_globus.so.$(VERSION): grst_err_globus.o \
          grst_err_globus.o grst_x509_globus.o grst_gacl_globus.o grst_xacml_globus.o \
          grst_http_globus.o grst_asn1_globus.o
 	ln -sf libgridsite_globus.so.$(VERSION) libgridsite_globus.so
+	gcc -shared -Wl,-soname,libgridsite_globus.so.1.5 \
+         -o libgridsite_globus.so.1.5 $(XML2_LIBS) \
+         grst_err_globus.o grst_x509_globus.o grst_gacl_globus.o grst_xacml_globus.o \
+         grst_http_globus.o grst_asn1_globus.o
 
 libgridsite_globus.a: grst_err_globus.o grst_x509_globus.o grst_gacl_globus.o grst_http_globus.o grst_asn1_globus.o
 	ar src libgridsite_globus.a \
@@ -179,6 +187,7 @@ else
 
 libgridsite_globus.so.$(VERSION): libgridsite.so.$(VERSION)
 	cp -f libgridsite.so.$(VERSION) libgridsite_globus.so.$(VERSION)
+	cp -f libgridsite.so.1.5 libgridsite_globus.so.1.5
 
 libgridsite_globus.a: libgridsite.a
 	cp -f libgridsite.a libgridsite_globus.a
@@ -195,6 +204,10 @@ libgridsite_nossl.so.$(VERSION): grst_err_nossl.o \
          grst_err_nossl.o grst_gacl_nossl.o grst_xacml_nossl.o \
          grst_http_nossl.o grst_htcp_nossl.o
 	ln -sf libgridsite_nossl.so.$(VERSION) libgridsite_nossl.so
+	gcc -shared -Wl,-soname,libgridsite_nossl.so.1.5 \
+         -o libgridsite_nossl.so.1.5 $(XML2_LIBS) \
+         grst_err_nossl.o grst_gacl_nossl.o grst_xacml_nossl.o \
+         grst_http_nossl.o grst_htcp_nossl.o
 
 libgridsite_nossl.a: grst_err_nossl.o grst_gacl_nossl.o grst_http_nossl.o
 	ar src libgridsite_nossl.a \
@@ -467,6 +480,9 @@ install-lib:
                                  $(prefix)/$(libdir)/libgridsite_nossl.so.$(MAJOR_VERSION)
 	ln -sf libgridsite_nossl.so.$(PATCH_VERSION) \
                                  $(prefix)/$(libdir)/libgridsite_nossl.so.$(MINOR_VERSION)
+	cp -f  libgridsite.so.1.5 $(prefix)/$(libdir)
+	cp -f  libgridsite_globus.so.1.5 $(prefix)/$(libdir)
+	cp -f  libgridsite_nossl.so.1.5 $(prefix)/$(libdir)
 
 install-slashgrid: slashgrid
 	cp -f slashgrid $(prefix)/sbin
diff --git a/org.gridsite.core/src/make-gridsite-spec b/org.gridsite.core/src/make-gridsite-spec
index f7821e0..8c2c7a5 100755
--- a/org.gridsite.core/src/make-gridsite-spec
+++ b/org.gridsite.core/src/make-gridsite-spec
@@ -53,6 +53,7 @@ Summary: GridSite
 License: Modified BSD
 Group: System Environment/Daemons
 Source: %{name}-%{version}.src.tar.gz
+Patch1: compat-1.5.patch
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root
 URL: http://www.gridsite.org/
 Vendor: GridPP
@@ -314,6 +315,19 @@ See http://www.gridsite.org/ for details.
 %attr(-, root, root) %{_prefix}/share/doc/gridsite-${MINOR_VERSION:-1.x}/htproxyunixtime.1
 %attr(-, root, root) %{_prefix}/share/doc/gridsite-${MINOR_VERSION:-1.x}/htproxyrenew.1
 %attr(-, root, root) %{_prefix}/share/doc/gridsite-${MINOR_VERSION:-1.x}/htproxyinfo.1
+
+%package 1.5-compat
+Group: Development/Libraries
+Summary: GridSite 1.5 compatibility package
+Requires: gridsite-shared
+
+%description 1.5-compat
+GridSite 1.5 compatibility package.
+
+%files 1.5-compat
+%attr(-, root, root) %{_prefix}/%{_lib}/libgridsite.so.1.5
+%attr(-, root, root) %{_prefix}/%{_lib}/libgridsite_globus.so.1.5
+%attr(-, root, root) %{_prefix}/%{_lib}/libgridsite_nossl.so.1.5
 EOF
 
 fi
